### PR TITLE
Gem/demo data

### PR DESCRIPTION
**Pull request by nkadunc**
Notebook with eo-data examples

### CODE REVIEW

File: GEM-data/eo-commercial-data.ipynb [hunk 1]
```diff
@@ -0,0 +1,539 @@
...
+   "source": [
+    "ps_data = DataCollection.define_byoc(\n",
+    "    \"906b545b-4305-4968-b245-2abcb0609bc1\",\n",
+    "    service_url=\"https://services.sentinel-hub.com\",\n",
+    "    bands=[\n",
+    "        Band(name=\"Blue\", units=(Unit.DN,), output_types=(np.uint16)),\n",
+    "        Band(name=\"Green\", units=(Unit.DN,), output_types=(np.uint16)),\n",
+    "        Band(name=\"Red\", units=(Unit.DN,), output_types=(np.uint16)),\n",
+    "        Band(name=\"NIR\", units=(Unit.DN,), output_types=(np.uint16)),\n",
+    "    ],\n",
```
Comment (zigaLuksic) [inline GEM-data/eo-commercial-data.ipynb:98]: Here you could also do
```python
band_names = ["Blue", "Green", "Red", "NIR"]
[ Band(name=name, units=(Unit.DN,), output_types=(np.uint16,)) for name in band_names)]
```
I also think you're missing a comma in `(np.uint16,)`.
This applies to other occurrences as well.

File: GEM-data/introduction.ipynb [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+   "outputs": [],
+   "source": [
+    "config.save()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### meteoblue account\n",
```
Comment (zigaLuksic) [inline GEM-data/introduction.ipynb:74]: Should this be capitalized?

File: GEM-data/weather-data.ipynb [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "eopatch = EOPatch()\n",
+    "eopatch.bbox = to_utm_bbox(coast_bbox)\n",
+    "eopatch.timestamp = [datetime.datetime(2022, 8, 1, 11, 0, 0)]"
```
Comment (zigaLuksic) [inline GEM-data/weather-data.ipynb:76]: Instead try:
```python
eopatch = EOPatch(bbox=to_utm_bbox(coast_bbox), timestamp=datetime.datetime(...))
```

Also applies later on.

**Review by zigaLuksic — COMMENTED** [review]
Can't tag all notebooks directly (diffs too large) so I'll add some remarks here.

It would be great if plotting were done with `EOPatch.plot` when possible (see [examples](https://github.com/sentinel-hub/eo-learn/blob/master/examples/visualization/EOPatchVisualization.ipynb)

For `eo-data.ipynb`:
- when downloading all of the bands (I think you do that once or twice), you can leave the `bands` parameter as `None` (default)

For `eo-derived-data.ipynb`:
- you can again specify bands with a list comprehension
- For the multi-value-enums it would be great if the docstring would describe what each value represents
- The procedure to extract cmap and cnorm repeats a couple of time, you should extract it to a function. Another option is to do it similarly to how it's done for the SCL multi-value-enum
- same goes for the plot procedure
- there is an empty cell somewhere in the notebook that can be removed

**Comment by batic** [comment]
I believe things are mostly fixed. I've skipped flake8 checking for too long lines (don't judge), and did some small refactoring wrt codelists for various derived data collections. Where easy to swap, I've used eopatch plotting utilities.

**Review by zigaLuksic — APPROVED** [review]
In the `eo-derived-data` you could probably still extract the plotting to a function, but I'll leave that decision to you.

Another minor complaint is that the notebooks use `-` for spacing while the utility python file uses  `_`.

But those are both minor, so merge at will

**Pull request merged**